File: src/main/java/CodingExam/BaekJoon/Exam25304.java
package CodingExam.BaekJoon;

import java.util.Scanner;

public class Exam25304 {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int value = sc.nextInt(); // 총 값
        int total = sc.nextInt(); // 총 개수

        int[] arrValue = new int[total];
        int[] arrTotal = new int[total];
        for (int i = 0; i < total; i++) {
            arrValue[i] = sc.nextInt();
            arrTotal[i] = sc.nextInt();
        }

        int sum = 0;
        for (int i = 0; i < total; i++) {
            sum += arrValue[i] * arrTotal[i];
        }

        System.out.println("sum = " + sum);
        if(value == sum)
            System.out.println("Yes");
        else
            System.out.println("No");

    }
}

File: src/main/java/CodingExam/BaekJoon/Exam25304.py
X = int(input())
N = int(input())

# total = 0
#
# for _ in range(N):
#     price, quantity = map(int, input().split())
#     total += price * quantity
#
# if total == X:
#     print("Yes")
# else:
#     print("No")
#

total = sum(int(price) * int(quantity) for price, quantity in (input().split() for _ in range(N)))

print("Yes" if total == X else "No")
